mesonbuild/build.py

```diff
@@ -1236,6 +1239,15 @@ def process_kwargs(self, kwargs):
             if self.gnu_symbol_visibility not in permitted:
                 raise InvalidArguments('GNU symbol visibility arg {} not one of: {}'.format(self.gnu_symbol_visibility, ', '.join(permitted)))
 
+        if 'rust_dependency_map' in kwargs:
+            rust_dependency_map = kwargs['rust_dependency_map']
+            if isinstance(rust_dependency_map, dict):
+                if any(not isinstance(v, str) for v in rust_dependency_map.values()):
+                    raise InvalidArguments(f'Invalid rust_dependency_map "{rust_dependency_map}": must be a dictionary with string values.')
+                self.rust_dependency_map = rust_dependency_map
+            else:
+                raise InvalidArguments(f'Invalid rust_dependency_map "{rust_dependency_map}": must be a dictionary.')
```

Comment:
```suggestion
        self.rust_dependency_map = kwargs.get('rust_dependency_map', {})
        if not isinstance(self.rust_dependency_map, dict):
            raise InvalidArguments(f'rust_dependency_map must be a dictionary.')
        if any(not isinstance(v, str) for v in self.rust_dependency_map.values()):
            raise InvalidArguments(f'rust_dependency_map must be a dictionary with string values.')
```

That avoids a double lookup in the kwargs dict and means self.rust_dependency_map is always defined so you can remove the hasattr in post_init(). Also not sure it's safe to have `{rust_dependency_map}` in the exception message, because that could be anything that might not be convertible to string.

Reply:
Changed accordingly, but this of course means that it's now always defined even if the target is not a Rust target. During the kwargs processing, `self.compilers` is not set correctly yet but only later in `post_init`. Similar problem as in https://github.com/mesonbuild/meson/pull/11698.

Reply:
I would prefer it to be always defined, having conditionally defined class attributes is generally considered a code-smell in python.